apps/set-batch-overrides/src/set_batch_overrides.py

```diff
@@ -0,0 +1,86 @@
+AUTORIFT_S2_MEMORY = '7875'
+AUTORIFT_LANDSAT_MEMORY = '15750'
+RTC_GAMMA_10M_MEMORY = '63200'
+WATER_MAP_10M_MEMORY = '126000'
+
+INSAR_ISCE_BURST_MEMORY_8G = '7500'
+INSAR_ISCE_BURST_MEMORY_16G = '15500'
+INSAR_ISCE_BURST_MEMORY_32G = '31500'
+INSAR_ISCE_BURST_MEMORY_64G = '63500'
+INSAR_ISCE_BURST_MEMORY_128G = '127500'
+
+# vCPU = Memory/8 for r6 instance types
+INSAR_ISCE_BURST_OMP_NUM_THREADS = {
+    INSAR_ISCE_BURST_MEMORY_8G: '1',
+    INSAR_ISCE_BURST_MEMORY_16G: '2',
+    INSAR_ISCE_BURST_MEMORY_32G: '4',
+    INSAR_ISCE_BURST_MEMORY_64G: '8',
+    INSAR_ISCE_BURST_MEMORY_128G: '16',
+}
+
+
+def get_container_overrides(memory: str, omp_num_threads: str = None) -> dict:
+    container_overrides = {
+        'ResourceRequirements': [
+            {
+                'Type': 'MEMORY',
+                'Value': memory,
+            }
+        ]
+    }
+    if omp_num_threads is not None:
+        container_overrides['Environment'] = [{'Name': 'OMP_NUM_THREADS', 'Value': omp_num_threads}]
+    return container_overrides
+
+
+def get_insar_isce_burst_memory(job_parameters: dict) -> str:
+    looks = job_parameters['looks']
+    bursts = len(job_parameters['reference'].split(' '))
+    if looks == '5x1':
+        if bursts < 2:
+            return INSAR_ISCE_BURST_MEMORY_8G
+        if bursts < 4:
+            return INSAR_ISCE_BURST_MEMORY_16G
+        if bursts < 11:
+            return INSAR_ISCE_BURST_MEMORY_32G
+        if bursts < 25:
+            return INSAR_ISCE_BURST_MEMORY_64G
+        if bursts < 31:
+            return INSAR_ISCE_BURST_MEMORY_128G
+    if looks == '10x2':
+        if bursts < 8:
+            return INSAR_ISCE_BURST_MEMORY_8G
+        if bursts < 21:
+            return INSAR_ISCE_BURST_MEMORY_16G
+        if bursts < 31:
+            return INSAR_ISCE_BURST_MEMORY_32G
+    if looks == '20x4':
+        if bursts < 23:
+            return INSAR_ISCE_BURST_MEMORY_8G
+        if bursts < 31:
+            return INSAR_ISCE_BURST_MEMORY_16G
+    raise ValueError(f'No memory value for {bursts} bursts and {looks} looks')
```

Comment:
This probably won't change too much, so it may not matter, but this would be more extensible and clean if the limits were in a dict. Something like:

```python
if looks == '5x1':
  burst_limits_5x1 = {
      2: INSAR_ISCE_BURST_MEMORY_8G,
      4: INSAR_ISCE_BURST_MEMORY_16G, 
      11: INSAR_ISCE_BURST_MEMORY_32G,
      25: INSAR_ISCE_BURST_MEMORY_64G,
      31: INSAR_ISCE_BURST_MEMORY_128G
  }
  mem_req_key = min(burst_limits_5x1, key=lambda x: abs(x-bursts))
  return burst_limits_5x1[mem_req_key]
if looks == '10x2':
  ...
``

Reply:
Yeah, I had a full implementation of something similar, using a data structure (I think a list of intervals) rather than a bunch of `if` statements, but in the end I decided it was actually less readable and flexible. Feel free to take a stab at a refactor if you're inspired, but I'm content with the current version.